### PR TITLE
Built an auto balanced system for prefab spawning

### LINKED ISSUE #1779: Scannable objects do not spawn (probability too low)
### Version
 1.6.0.0

### Bug
 All scanable objects don't appear on the map. It's a very big issue because we can't have the modication station or other things that we need to progress in the game.

### PR DESCRIPTION
**Pull request by tornac1234**
Hotfix for the current issue of player never finding most of the scannable objects spawning around wrecks for example

Fixes #1779

### CODE REVIEW

File: NitroxServer/GameLogic/Entities/EntityHelper.cs [hunk 1]
```diff
@@ -0,0 +1,8 @@
+﻿using NitroxModel.DataStructures.GameLogic;
+
+namespace NitroxServer.GameLogic.Entities;
+
+public abstract class EntityHelper
```
Comment (Jannify) [inline NitroxServer/GameLogic/Entities/EntityHelper.cs:5]: Isn't an interface more fitting here as it requires to implement the method?

File: NitroxServer-Subnautica/GameLogic/Entities/SubnauticaEntityHelper.cs [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+public class SubnauticaEntityHelper : EntityHelper
+{
+    public override bool TryGetTechTypeForClassId(string classId, out NitroxTechType nitroxTechType)
+    {
+        ResourceAssets resourceAssets = ResourceAssetsParser.Parse();
+        if (resourceAssets.WorldEntitiesByClassId.TryGetValue(classId, out UWE.WorldEntityInfo worldEntityInfo))
+        {
+            nitroxTechType = worldEntityInfo.techType.ToDto();
+            // At this point, the tech type can still be None, so we need to make sure we return true only if we have a valid tech type
+            return nitroxTechType.ToUnity() != TechType.None;
```
Comment (Jannify) [inline NitroxServer-Subnautica/GameLogic/Entities/SubnauticaEntityHelper.cs:17]: Can't we use `worldEntityInfo.techType != TechType.None` so we don't have a double cast?

File: NitroxServer/GameLogic/Entities/Spawning/BatchEntitySpawner.cs [hunk 3]
```diff
@@ -28,6 +32,14 @@ public class BatchEntitySpawner : IEntitySpawner
         private readonly object emptyBatchesLock = new object();
         private HashSet<NitroxInt3> parsedBatches;
 
+        /// <summary>
+        /// Should not be modified without testing (a good way is "goto wreck13" and look around the wreck)
+        /// 0.1 seemed to be a bit low and 0.3 seemed to be too much
+        /// </summary>
+        private const float FRAGMENT_LUCK_INDICATOR = 0.2f;
+        private const float DEFAULT_LUCK_INDICATOR = 0.1f;
```
Comment (Jannify) [inline NitroxServer/GameLogic/Entities/Spawning/BatchEntitySpawner.cs:40]: As discussed in discord this is not a valid option as the "luck indicator" changes with progression and fragment type afair.

**Review by Jannify — CHANGES_REQUESTED** [review]

**Comment by dartasen** [comment]
Since the issue is fixed, we can close this one for now

**Pull request closed without merging**